src/buffer/out/textBuffer.cpp

```diff
@@ -2533,17 +2559,36 @@ void TextBuffer::_AppendRTFText(std::string& contentBuilder, const std::wstring_
     }
 }
 
-void TextBuffer::Serialize(const wchar_t* destination) const
+void TextBuffer::SerializeToPath(const wchar_t* destination) const
 {
     const wil::unique_handle file{ CreateFileW(destination, GENERIC_WRITE, FILE_SHARE_READ | FILE_SHARE_DELETE, nullptr, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, nullptr) };
     THROW_LAST_ERROR_IF(!file);
 
     static constexpr size_t writeThreshold = 32 * 1024;
+    ChunkedSerialize(writeThreshold, L"\uFEFF", nullptr, [&](std::wstring& buffer, bool /* isDone */) {
+        const auto fileSize = gsl::narrow<DWORD>(buffer.size() * sizeof(wchar_t));
+        DWORD bytesWritten = 0;
+        THROW_IF_WIN32_BOOL_FALSE(WriteFile(file.get(), buffer.data(), fileSize, &bytesWritten, nullptr));
+        THROW_WIN32_IF_MSG(ERROR_WRITE_FAULT, bytesWritten != fileSize, "failed to write");
+        buffer.clear();
+    });

```

Comment:
Possible refactoring:
I think we can avoid `std::function` usage here, by making `ChunkedSerialize` chunk buffer in row-by-row fashion instead of bytes-sized data. Then, both `SerializeToPath` and `GetWithControlSequences` can decide to read more or flush based on their specific logic.

This might be more complicated, so it's okay to not delve into this right now.

Reply:
Also, if you decide to do this, then rename `ChunkedSerialize` to `SerializeRow` to be explicit.